helix-term/src/commands.rs

```diff
@@ -895,6 +896,43 @@ pub fn append_mode(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+// Creates an LspCallback that waits for formatting changes to be computed. When they're done,
+// it applies them, but only if the doc version hasn't changed.
+fn make_format_callback(
+    doc_id: DocumentId,
+    doc_version: i32,
+    set_unmodified: bool,
+    format: impl Future<Output = Arc<Mutex<anyhow::Result<Transaction>>>> + Send + 'static,
+) -> LspCallback {
+    Box::pin(async move {
+        let trans = format.await;
+        let call: LspCallbackWrapper =
+            Box::new(move |editor: &mut Editor, compositor: &mut Compositor| {
+                let view_id = editor.view().id;
+                if let Some(doc) = editor.document_mut(doc_id) {
+                    if doc.version() == doc_version {
```

Comment:
If the version changed, we could later on use `Transaction::map`. It's currently unimplemented because I didn't need it yet:

https://github.com/helix-editor/helix/blob/a3cb79ebaaaa7682c7c7b007ed928b24f80cf3c2/helix-core/src/transaction.rs#L253-L266

Reply:
That sounds useful, yes! Will there be a way to check if the two transactions "interact/conflict"? I worry a little about messing up the user's edits, and given that we'll reformat anyway at the next save I think it's usually fine to just throw away the reformatting.